### PR TITLE
Add dummy server

### PR DESCRIPTION
**Pull request by iomodo**
This PR adds dummy server.
`go run ./cmd/main.go` will run the server

### CODE REVIEW

File: core/kg/store/sqlstore/user_store.go [hunk 1]
```diff
@@ -13,6 +13,8 @@ type SqlUserStore struct {
 	usersQuery sq.SelectBuilder
 }
 
+var _ store.UserStore = SqlUserStore{}
```
Comment (giolekva) [inline core/kg/store/sqlstore/user_store.go:16]: What's this for?
To typecheck if SqlUserStore impelements store.UserStore?

File: core/kg/server/server.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	}
+
+	a.srv = grpc.NewServer()
+
+	a.Log.Info("Server is listening on", log.String("port", listenerPort))
+	if err := a.srv.Serve(lis); err != nil {
+		a.Log.Error("failed to serve: %v", log.Err(err))
+		return err
+	}
+	a.Log.Info("Server stopped")
```
Comment (giolekva) [inline core/kg/server/server.go:50]: s/stopped/started/

File: core/kg/server/server.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	return nil
+}
+
+// Shutdown method shuts server down
+func (a *Server) Shutdown() {
+	a.Log.Info("Stoping Server...")
```
Comment (giolekva) [inline core/kg/server/server.go:56]: Stopping

File: core/kg/server/server.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	// settings := model.NewConfig().SqlSettings
+	// a.store = sqlstore.New(settings)
+
+	lis, err := net.Listen("tcp", listenerPort)
+	if err != nil {
+		a.Log.Error("failed to listen: %v", log.Err(err))
+		return err
+	}
+
+	a.srv = grpc.NewServer()
```
Comment (giolekva) [inline core/kg/server/server.go:43]: Should grpc server be passed from outside as well?
Will that help during testing (to pass mock implementation)?

File: core/kg/log/log.go [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+			Compress: true,
+		})
+		core := zapcore.NewCore(makeEncoder(config.FileJSON), writer, logger.fileLevel)
+		cores = append(cores, core)
+	}
+
+	combinedCore := zapcore.NewTee(cores...)
+
+	logger.zap = zap.New(combinedCore,
+		zap.AddCaller(),
```
Comment (giolekva) [inline core/kg/log/log.go:77]: I'd move creation of logger variable from above to here, initialize all three member attributes together and return logger right away

File: core/kg/log/log.go [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+	}
+}
+
+func makeEncoder(json bool) zapcore.Encoder {
+	encoderConfig := zap.NewProductionEncoderConfig()
+	if json {
+		return zapcore.NewJSONEncoder(encoderConfig)
+	}
+
+	encoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
```
Comment (giolekva) [inline core/kg/log/log.go:120]: Question: this is for consistently formatting datetime strings right?
And we skip it when using json to keep int64s

File: core/kg/server/server.go [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+// NewServer creates new Server
+func NewServer(logger *log.Logger) (*Server, error) {
+	a := &Server{
+		Log: logger,
+	}
+
+	pwd, _ := os.Getwd()
+	a.Log.Info("Printing current working", log.String("directory", pwd))
```
Comment (giolekva) [inline core/kg/server/server.go:28]: Just "Current working directory: %s" ?

**Review by giolekva — APPROVED** [review]

File: core/kg/store/sqlstore/user_store.go (hunk 1 above)
Comment (iomodo) [reply core/kg/store/sqlstore/user_store.go:16]: Correct

**Pull request merged**